OpenRA.Game/FileSystem/InstallShieldCABExtractor.cs

```diff
@@ -404,9 +405,44 @@ public string DirectoryName(uint index)
 			return test;
 		}
 
+		public string GetFirstSubDirectory(string package)
```

Comment:
What is the purpose of this method?  The implementation looks to be DIYing some path manipulation that the standard library should provide methods for.